recipes/unix.rb

```diff
@@ -0,0 +1,190 @@
+# THE SOFTWARE.
+
+setup_root       = node['android-sdk']['setup_root'].to_s.empty? ? node['ark']['prefix_home'] : node['android-sdk']['setup_root']
+android_home     = File.join(setup_root, node['android-sdk']['name'])
+android_bin      = File.join(android_home, 'tools', 'android')
+
+#
+# Install required libraries
+#
+if node['platform'] == 'ubuntu'
+  package 'libgl1-mesa-dev'
+
+  #
+  # Install required 32-bit libraries on 64-bit platforms
+  #
+  if node['kernel']['machine'] != 'i686'
+    # http://askubuntu.com/questions/147400/problems-with-eclipse-and-android-sdk
+    if Chef::VersionConstraint.new('>= 13.04').include?(node['platform_version'])
+      package 'lib32stdc++6'
+    elsif Chef::VersionConstraint.new('>= 11.10').include?(node['platform_version'])
+      package 'libstdc++6:i386'
+    end
+
+    package 'lib32z1'
+  end
+end
+
+#
+# Download and setup android-sdk tarball package
+#
+if linux?
+  sdk_url = node['android-sdk']['download_url']['linux']
+  sdk_checksum = node['android-sdk']['checksum']['linux']
+elsif mac_os_x?
+  sdk_url = node['android-sdk']['download_url']['mac_os_x']
+  sdk_checksum = node['android-sdk']['checksum']['mac_os_x']
+else
+  Chef::Log.error('Platform not supported!')
+end
+
+ark node['android-sdk']['name'] do
+  url sdk_url
+  path node['android-sdk']['setup_root']
+  checksum sdk_checksum
+  version node['android-sdk']['version']
+  prefix_root node['android-sdk']['setup_root']
+  prefix_home node['android-sdk']['setup_root']
+  owner node['android-sdk']['owner']
+  group node['android-sdk']['group'] unless mac_os_x?
+  backup node['android-sdk']['backup_archive']
+  action node['android-sdk']['with_symlink'] ? :install : :put
+end
+
+#
+# Fix non-friendly 0750 permissions in order to make android-sdk available to all system users
+#
+%w(add-ons platforms tools).each do |subfolder|
+  directory File.join(android_home, subfolder) do
+    mode '0755'
+  end
+end
+# TODO: find a way to handle 'chmod stuff' below with own chef resource (idempotence stuff...)
+execute 'Grant all users to read android files' do
+  command "chmod -R a+r #{android_home}/*"
+  user node['android-sdk']['owner']
+  group node['android-sdk']['group'] unless mac_os_x?
+end
+execute 'Grant all users to execute android tools' do
+  command "chmod -R a+X #{File.join(android_home, 'tools')}/*"
+  user node['android-sdk']['owner']
+  group node['android-sdk']['group'] unless mac_os_x?
+end
+
+#
+# Configure environment variables (ANDROID_HOME and PATH)
+#
+if linux?
+  template "/etc/profile.d/#{node['android-sdk']['name']}.sh" do
+    source 'android-sdk.sh.erb'
+    mode '0644'
+    owner node['android-sdk']['owner']
+    group node['android-sdk']['group'] unless mac_os_x?
+    variables(
+      android_home: android_home
+    )
+    only_if { node['android-sdk']['set_environment_variables'] }
+  end
+elsif mac_os_x?
+  bash_profile 'profile.android_sdk' do
```

Comment:
Yup, that's what I was doing. In simple kitchen test it's fine, it's when it's integrated with an environment cookbook. More specifically, when logging in as one user, then sudoing to another. I actually think the `export PATH` online 116 could probably go in `/etc/paths.d/android-sdk` then it's system wide, not just to specific user.